### PR TITLE
feat: add cicd and update openmldb version

### PR DESCRIPTION
**Pull request by tobegit3hub**
* Add cicd.yml to build spark distribution
* Upgrade openmldb to 0.4.0-snapshot
* Add snapshot repository in pom.xml

### CODE REVIEW

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: cicd
+
+on:
+  push:
+    branches:
+      - v3.0.0-openmldb
+  pull_request:
```
Comment (aceforeverd) [inline .github/workflows/cicd.yml:7]: add `workflow_dispatch` so we can run manually

File: .github/workflows/cicd.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('java/**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: compile
+        run: |
+          ./dev/make-distribution.sh --name openmldbspark --pip --tgz -Phadoop-2.7 -Pyarn
```
Comment (aceforeverd) [inline .github/workflows/cicd.yml:33]: missing release step

File: .github/workflows/cicd.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: cicd
+
+on:
+  push:
+    branches:
+      - v3.0.0-openmldb
```
Comment (aceforeverd) [inline .github/workflows/cicd.yml:6]: add tag push condition ?

File: .github/workflows/cicd.yml (hunk 2 above)
Comment (tobegit3hub) [reply .github/workflows/cicd.yml:33]: We will add release later, maybe in another job.

File: .github/workflows/cicd.yml (hunk 3 above)
Comment (tobegit3hub) [reply .github/workflows/cicd.yml:6]: We will add this later.

File: .github/workflows/cicd.yml (hunk 1 above)
Comment (tobegit3hub) [reply .github/workflows/cicd.yml:7]: okay.

**Pull request merged**